zopkio/test_runner.py

```diff
@@ -198,25 +251,15 @@ def _copy_logs(self):
     """
     Copy logs from remote machines to local destination
     """
-    if "LOGS_DIRECTORY" in self.master_config.mapping:
-      logs_dir = self.master_config.mapping.get("LOGS_DIRECTORY")
-    else:
-      logs_dir = self.dynamic_config_module.LOGS_DIRECTORY
-    utils.makedirs(logs_dir)
+
     for deployer in runtime.get_deployers():
       for process in deployer.get_processes():
-        logs = []
-        if (hasattr(self.dynamic_config_module, "process_logs")):
-          logs += self.dynamic_config_module.process_logs(process.servicename)
-        if (hasattr(self.dynamic_config_module, "machine_logs")):
-          logs += self.dynamic_config_module.machine_logs().get(process.unique_id, [])
-        if (hasattr(self.dynamic_config_module, "naarad_logs")):
-          logs += self.dynamic_config_module.naarad_logs().get(process.unique_id, [])
-        if hasattr(self.dynamic_config_module, 'log_patterns'):
-          pattern = self.dynamic_config_module.log_patterns().get(process.unique_id, '^$')
-        else:
-          pattern = '^$'
-        deployer.get_logs(process.unique_id, logs, logs_dir, pattern)
+        logs = self.dynamic_config_module.process_logs( process.servicename) or []
+        logs += self.dynamic_config_module.machine_logs( process.unique_id)
```

Comment:
The __init__ (constructor) should guarantee this method exists.  It will
define it as returning an empty list if it is not there.

On another note, I am assuming that the preferred way of running at this
point is using ZTestSuite which has this method, but this is meant to be
backwards compatible with (deprecated?) workflow.  Perhaps this isn't quite
the right view, so advise if not.

The intent here was to clean up the logic here and move the complexity into
the constructor.  It made it easier to deal with the discrepancy now in
(deprecated?) machine_logs that was defined not to take a parameter vs one
that does take a parameter.

With that said, this approach may need a set of eyes to know if there is
potential for conflict other issues I haven't foreseen.

-John


On Thu, Sep 17, 2015 at 2:44 PM, Joshua Ehrlich <notifications@github.com>
wrote:

> In zopkio/test_runner.py
> <https://github.com/linkedin/Zopkio/pull/95#discussion_r39803026>:
>
> >      for deployer in runtime.get_deployers():
> >        for process in deployer.get_processes():
> > -        logs = []
> > -        if (hasattr(self.dynamic_config_module, "process_logs")):
> > -          logs += self.dynamic_config_module.process_logs(process.servicename)
> > -        if (hasattr(self.dynamic_config_module, "machine_logs")):
> > -          logs += self.dynamic_config_module.machine_logs().get(process.unique_id, [])
> > -        if (hasattr(self.dynamic_config_module, "naarad_logs")):
> > -          logs += self.dynamic_config_module.naarad_logs().get(process.unique_id, [])
> > -        if hasattr(self.dynamic_config_module, 'log_patterns'):
> > -          pattern = self.dynamic_config_module.log_patterns().get(process.unique_id, '^$')
> > -        else:
> > -          pattern = '^$'
> > -        deployer.get_logs(process.unique_id, logs, logs_dir, pattern)
> > +        logs = self.dynamic_config_module.process_logs( process.servicename) or []
> > +        logs += self.dynamic_config_module.machine_logs( process.unique_id)
>
> will this fail if there is no function called machine_logs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/linkedin/Zopkio/pull/95/files#r39803026>.
>



-- 
John Rusnak
Sr. Software Engineer
LinkedIn